fdbserver/DataDistribution.actor.cpp

```diff
@@ -911,6 +920,122 @@ ACTOR static Future<Void> waitForMost(std::vector<Future<ErrorOr<Void>>> futures
 	return Void();
 }
 
+ACTOR Future<std::map<NetworkAddress, std::pair<WorkerInterface, std::string>>> getStatefulWorkers(
```

Comment:
It looks like this function is duplicating code from `getStorageWorkers` and `getDatabaseConfiguration`. Can you remove this duplicate code? For `getDatabaseConfiguration`, it looks like we should add a new `getDatabaseConfiguration(Transaction const &tr)` method to the management API.

Reply:
I added a transaction version func of `getDatabaseConfiguration`

`getStorageWorkers` is used by another workload, `FailureDiskInjection`, so may not remove it.
Here we are basically using the transaction version of it but combined with `getWorkers` and the return type are also not the same.
It's a bit complex to combine them and reuse the interface.
I guess we can keep it like this for now?